### PR TITLE
GROOVY-9103: CLONE - CLONE - Fix warning "An illegal reflective acces…

### PR DESCRIPTION
**Pull request by danielsun1106**
…s operation has occurred"

### CODE REVIEW

File: src/main/java/groovy/lang/MetaClassImpl.java [hunk 1]
```diff
@@ -2207,11 +2209,28 @@ private static MetaMethod getCategoryMethodSetter(Class sender, String name, boo
 //                    element = new MetaBeanProperty(mp.getName(), mp.getType(), null, mp.getSetter());
 //                }
             }
+            if (element instanceof MetaBeanProperty) {
+                MetaBeanProperty mbp = (MetaBeanProperty) element;
+                boolean getterAccessible = canAccess(mbp.getGetter());
+                boolean setterAccessible = canAccess(mbp.getSetter());
+
+                if (!(getterAccessible && setterAccessible)) continue;
```
Comment (danielsun1106) [inline src/main/java/groovy/lang/MetaClassImpl.java]: @paulk-asert Could you review the logic of fitering? 
getter and setter of property should be access legally, or the property will be filtered for now.
Also, filtering non-properties is a breaking change. Should we provide an option to allow the illegal access?(Suggestion: forbiddening illegal access by default)

**Comment by danielsun1106** [comment]
JVM option `groovy.allow.illegal.access.properties` is added to enable illegal access properties for backward compatibility(Note: `false` is default value)

**Comment by danielsun1106** [comment]
Search "WARNING: Illegal reflective access by" in the Java11 build log( https://api.travis-ci.org/v3/job/537214690/log.txt ), only 5 illegal access warning remains. They are all truely illegal access, we have to fix them by changing our code.

**Comment by danielsun1106** [comment]
@paulk-asert The PR is ready to review :-)

**Comment by danielsun1106** [comment]
@paulk-asert  ping  ;)

File: src/main/java/groovy/lang/MetaClassImpl.java [hunk 2]
```diff
@@ -2207,11 +2210,34 @@ private static MetaMethod getCategoryMethodSetter(Class sender, String name, boo
...
+                    if (!(getterAccessible && setterAccessible)) continue;
+                }
+            }
+
             ret.add(element);
         }
         return ret;
     }
 
+    private static final boolean ALLOW_ILLEGAL_ACCESS_PROPERTIES = SystemUtil.getBooleanSafe("groovy.allow.illegal.access.properties");
```
Comment (paulk-asert) [inline src/main/java/groovy/lang/MetaClassImpl.java]: This all looks good - my only suspicion is that some folks will want to turn this feature on very selectively.
We could make an instance property which is set by the system property but could be toggled on an individual metaclass basis. I am also wondering whether "groovy.permissive.property.access" would be a better name. In future JDK versions the access might not be allowed at all - for us it is indicating that we will attempt to access the property.

Comment (danielsun1106) [reply src/main/java/groovy/lang/MetaClassImpl.java]: Thanks for you reviewing the PR, Paul :)

I wish I had caught the points:
1) Rename `groovy.allow.illegal.access.properties`  to `groovy.permissive.property.access`
2) Add an instance property whose value is initialized by the system property `groovy.permissive.property.access`, we use the instance property to control the illegal access to properties

**Comment by danielsun1106** [comment]
@paulk-asert I've refined the switch according to your suggestion, pls review the PR again :)

https://github.com/apache/groovy/pull/934/commits/8d61efba824f80fea546c16ee8646921455e2b48

**Pull request merged**